src/error.rs

```diff
@@ -0,0 +1,47 @@
+use httparse::Error as HttpParseError;
+use std::io::Error as IOError;
+use std::result;
+use thiserror::Error;
+
+// Alias for easier error handling and less boilerplate.
+pub type Result<T> = result::Result<T, AlcazarError>;
```

Comment:
Just gonna write to extract this to separate mod. You've done that already <3

Reply:
This is a pretty cool wrap. ;)